tests/integration/test_charm.py

```diff
@@ -0,0 +1,144 @@
+async def test_build_and_deploy(
+    ops_test: OpsTest, slurmrestd_charm: Coroutine[Any, Any, pathlib.Path], series: str
+    ) -> None:
+    """Deploy minimal working slurmrestd charm."""
+    res_slurmd = get_slurmd_res()
+    res_slurmctld = get_slurmctld_res()
+
+    charm = await slurmrestd_charm
+
+    await asyncio.gather(
+        # Fetch from charmhub slurmctld
+        ops_test.model.deploy(
+            SLURMCTLD,
+            application_name=SLURMCTLD,
+            channel="edge",
+            num_units=1,
+            resources=res_slurmctld,
+            series=series,
+        ),
+        ops_test.model.deploy(
+            SLURMD,
+            application_name=SLURMD,
+            channel="edge",
+            num_units=1,
+            resources=res_slurmd,
+            series=series,
+        ),
+        ops_test.model.deploy(
+            SLURMDBD,
+            application_name=SLURMDBD,
+            channel="edge",
+            num_units=1,
+            series=series,
+        ),
+        ops_test.model.deploy(
+            charm,
+            application_name=SLURMRESTD,
+            num_units=1,
+            series=series,
+        ),
+        ops_test.model.deploy(
+            "percona-cluster",
+            application_name="mysql",
+            channel="edge",
+            num_units=1,
+            series="bionic",
+        ),
+    )
+    
+    # Attach ETCD resource to the slurmctld controller
+    await ops_test.juju("attach-resource", SLURMCTLD, f"etcd={res_slurmctld['etcd']}")
+
+    # Add slurmdbd relation to slurmctld
+    await ops_test.model.relate(SLURMCTLD, SLURMDBD)
+
+    # Add mysql relation to slurmdbd
+    await ops_test.model.relate(SLURMDBD, "mysql")
+
+    # Add slurmctld relation to slurmrestd
+    await ops_test.model.relate(SLURMRESTD, SLURMCTLD)
+
+    # Attach NHC resource to the slurmd controller
+    await ops_test.juju("attach-resource", SLURMD, f"nhc={res_slurmd['nhc']}")
+
+    # Add slurmctld relation to slurmd
+    await ops_test.model.add_relation(SLURMD, SLURMCTLD)
+
+    # issuing dummy update_status just to trigger an event
+    async with ops_test.fast_forward():
+        await ops_test.model.wait_for_idle(apps=[SLURMRESTD], status="active", timeout=1000)
+        assert ops_test.model.applications[SLURMRESTD].units[0].workload_status == "active"
+
+
+@pytest.mark.abort_on_fail
+@tenacity.retry(
+    wait=tenacity.wait.wait_exponential(multiplier=2, min=1, max=30),
+    stop=tenacity.stop_after_attempt(3),
+    reraise=True,
+)
+async def test_munge_is_active(ops_test: OpsTest) -> None:
+    """Test that munge is active."""
+    logger.info("Checking that munge is active inside Juju unit...")
+    slurmctld_unit = ops_test.model.applications[SLURMCTLD].units[0]
+    res = (await slurmctld_unit.ssh("systemctl is-active munge")).strip("\n")
+    assert res == "active"
+
+
+# TODO: Currently there is a bug where slurmrestd can reach active status despite the
```

Comment:
Like I mentioned in the slurmd pull request, instead of TODO you should use IMPORTANT because you are calling out a code smell. Indicates there is a larger problem that we need to address!! Also, certain IDEs will block you from committing changes if there are unresolved TODO comments throughout the source code.

```suggestion
# IMPORTANT: Currently there is a bug where slurmrestd can reach active status despite the
```

Reply:
Did not know that about certain IDEs blocking changes with TODO :open_mouth:

Reply:
Yeah, PyCharm will block commits until all TODOs have been addressed. I find it rather nice especially if I forget to fix something.